33-room-occupancy-monitor/README.md

```diff
@@ -0,0 +1,113 @@
+
+The PIR motion sensor has a small plastic dome sticking out of the front of the reference node's housing. Since this sensor is integrated into the reference node already, there's no additional wiring required. When something moves in front of the sensor, it'll detect that motion and kick off logic in the firmware to send a "motion detected" note to Notehub.
+
+### Firmware
+
+Next, we need to flash the reference node with the room occupancy monitor firmware.
+
+1. Before we do anything, we need to pull in some dependencies for the firmware to work. After cloning this repository, run these commands: `git submodule update --init 33-room-occupancy-monitor/firmware/note-c` and `git submodule update --init 33-room-occupancy-monitor/firmware/sparrow-lora`. This will pull in the note-c and sparrow-lora submodules that this project depends on.
+1. There are a few ways to build and flash the firmware onto the reference node. These are covered in the [Sparrow Builder's Guide](https://dev.blues.io/sparrow/sparrow-builders-guide/). Follow the steps in that guide and then return to these instructions.
+1. Connect the STLINK-V3MINI to your development PC with a USB A to micro USB cable.
+1. Connect the STLINK to your reference node with the 2x7 JTAG ribbon cable.
+1. Build and flash the code using whichever method you selected when following the Sparrow Builder's Guide.
+1. Open a terminal emulator and connect to the STLINK's serial connection to view logs. See the documentation [here](https://dev.blues.io/sparrow/sparrow-builders-guide/#collecting-firmware-logs). 
+1. Start the program in debug mode (again, how you do this depends on the IDE: VS Code or STM32CubeIDE). In your terminal emulator's output, you should see something like this:
+
+```
+===================
+===================
+===== SPARROW =====
+===================
+Feb 16 2023 22:42:58
+2037335832365003001d001f
+APPLICATION HOST MODE
+CONSOLE TRACE ENABLED
+```
+
+You may also see a message like this:
+
+```
+pir: 1 motion events sensed
+```
+
+At startup, the PIR sensor may detect motion even if there is none. This is a one-off false positive that can be ignored, and the sensor should work as intended thereafter. If you encounter additional false positives, you may want to try increasing the sensor's threshold value, which makes the sensor less sensitive. Specifically, you would increase the `PIR_THRESHOLD` value in pir.c:
+
+```C
+#define PIR_THRESHOLD 100
```

Comment:
It would be nice if this were part of the app's configuration rather than having to dive into the implementation source.

Reply:
Or better still, overridable via an environment variable.

Reply:
Agree that environment variable would be best, but to keep it simple, I think a macro is ok for now. I could surround this with `#ifndef PIR_THRESHOLD/#endif` so that it can be overridden when CMake is invoked (e.g. via `CMAKE_C_FLAGS`), or did you have something else in mind?

Reply:
iirc, each app has a configuration structure that could be used to specify the PIR threshold in code. Then it's up to the developer if they want to add a hard-coded value in their app repo or pull it from a Notecard environment variable.

Reply:
> iirc, each app has a configuration structure that could be used to specify the PIR threshold in code.

That seems no better than a macro, no? Hard-coded either way and have to re-compile.

> or pull it from a Notecard environment variable.

I like the environment variable idea, but I do think it's going to be add a non-trivial amount of extra code. For the sake of speed, which is the priority at this stage, I'm going to go the overridable marcro route.

Reply:
I agree about the time perspective for environment variables, but I feel that real code is better than macros when thinking about the division of responsibility - what the app provides vs how it's used and configured, with the intent of making it obvious and simple for the consumer. The apps do have configuration structures which are a natural and clear way to express this. That said, I'm happy to concede.

Reply:
Discussed a solution w/ Mat on Slack, which is implemented with latest commit.